scan/lib/scan/runner.rb

```diff
@@ -135,27 +137,118 @@ def retryable_tests(input)
       return retryable_tests.uniq
     end
 
-    def handle_results(tests_exit_status)
-      if Scan.config[:disable_xcpretty]
-        unless tests_exit_status == 0
-          UI.test_failure!("Test execution failed. Exit status: #{tests_exit_status}")
+    def find_filename(type)
+      index = Scan.config[:output_types].split(',').index(type)
+      return nil if index.nil?
+      return (Scan.config[:output_files] || "").split(',')[index]
+    end
+
+    def output_html?
+      return Scan.config[:output_types].split(',').include?('html')
+    end
+
+    def output_junit?
+      return Scan.config[:output_types].split(',').include?('junit')
+    end
+
+    def output_json_compilation_database?
+      return Scan.config[:output_types].split(',').include?('json-compilation-database')
+    end
+
+    def output_html_filename
+      return find_filename('html')
+    end
+
+    def output_junit_filename
+      return find_filename('junit')
+    end
+
+    def output_json_compilation_database_filename
+      return find_filename('json-compilation-database')
+    end
+
+    def trainer_test_results
+      require "trainer"
+
+      results = {
+        number_of_tests: 0,
+        number_of_failures: 0,
+        number_of_retries: 0
+      }
+
+      result_bundle_path = Scan.cache[:result_bundle_path]
+
+      output_path = Scan.config[:output_directory] || Dir.mktmpdir
+      output_path = File.absolute_path(output_path)
+
+      UI.crash!("A -resultBundlePath is needed to parse the test results. This should not have happened. Please file an issue.") unless result_bundle_path
+
+      params = {
+        path: result_bundle_path,
+        silent: true
+      }
+
+      formatter = Scan.config[:xcodebuild_formatter].chomp
+
+      if output_html? && formatter != 'xcpretty'
+        UI.error("HTML output is only available with `xcodebuild_formatter: 'xcpretty'` right now...")
```

Comment:
nit: I'm getting this error even though I haven't explicitly chosen the html type, since it's in the default output_types. We can silence it by being explicit with just junit, but would it make sense to default that option according to the formatter?

Reply:
actually, I'm also seeing this when I don't have xcbeautify installed and xcpretty was selected by default as the xcodebuild_formatter (but no explicit formatter nor output_types choices made by me)

Reply:
Ahhhh, I'll remove this! I thought it would be helpful but seems like its probably more error prone that anything 🤷‍♂️